### ISSUE TITLE
[Bug]: Slow first search

### ISSUE THREAD

**Issue by Sinled** [chrisgrieser/shimmering-obsidian#148]
### Bug Description

Hi, thanks for this nice workflow, 
but i am having relatively big vault (~4000 files), and first search takes ages, like 15-20 seconds. Given that i usually search once or twice a day, almost all my searches take such amount of time.

Is it expected behaviour or something can be done about it? Alfred's native file search finds corresponding files instantly. 

PS. It seems there are newer version of workflow, but it is not in the Alfred gallery.

### Relevant Screenshot

<img width="618" alt="Screenshot 2023-09-12 at 11 44 46" src="https://github.com/chrisgrieser/shimmering-obsidian/assets/1402135/f2934026-16ba-4a6a-af2e-c8c13d0b91fa">

### Debugging Log

```Text
[11:43:48.138] Logging Started...
[11:43:52.824] Shimmering Obsidian[Script Filter] Queuing argument '(null)'
[11:44:08.441] Shimmering Obsidian[Script Filter] Script with argv '(null)' finished
[11:44:08.447] Shimmering Obsidian[Script Filter] {"items":[{"title":
```

### Workflow Configuration

<img width="511" alt="Screenshot 2023-09-12 at 11 57 04" src="https://github.com/chrisgrieser/shimmering-obsidian/assets/1402135/34446cf2-ba5e-4b16-a442-a5d6a372bacf">

### Version & System Information

```Text
-------------------------------
INTERNAL WORKFLOW CONFIGURATION
Vault Path: /Users/sinled/Library/Mobile Documents/iCloud~md~obsidian/Documents/main
config folder: exists
Metadata JSONs: 4/4
metadata.json String Length: 3936720
-------------------------------
WORKSPACE DATA
'workspace.json' exists
-------------------------------
SYSTEM
macOS: 13.5.2
Alfred: 5.1.2
-------------------------------
INSTALLED VERSION
Obsidian: 1.4.11
This Workflow: 4.0.4
Advanced URI Plugin: 1.38.1
Metadata Extractor: 1.1.0
-------------------------------
LATEST VERSION
Obsidian: 1.4.11 (Insider: 1.4.11)
This Workflow: 4.1.0
Advanced URI Plugin: 1.38.1
Metadata Extractor: 1.1.0
-------------------------------
```

### Make sure you have done the following

- [X] I have [updated to the latest version](https://github.com/chrisgrieser/shimmering-obsidian/releases/latest) of this workflow.
- [X] I have read the documentation of the feature I am having problems with.
- [X] I am using Alfred 5. (Alfred 4 is not supported by this workflow anymore.)
- [X] I have updated to the latest version of the Metadata-Extractor and the Advanced URI plugin.
- [X] If the `o…` main search does not work, I made sure that I haven't changed any of the default settings of the Metadata-Extractor. Furthermore, I have tried rerunning `osetup` and it still does not work.
- [X] If notes in general can be found, but recently created or renamed notes cannot, I used the `oupdate` command to manually update the metadata.

**Comment by chrisgrieser** [comment]
> PS. It seems there are newer version of workflow, but it is not in the Alfred gallery.

yeah, the Alfred Gallery unfortunately does not update instantly. Update should be in the gallery in the next few days. Or you can just update manually by downloading the latest release from github: https://github.com/chrisgrieser/shimmering-obsidian/releases/tag/4.1.0

---

Hmm, I have a similar number of notes and do not experience such a lag, it takes less than a second for me.

Could you elaborate what exactly you mean by "first search"? First search in a given day, in a given hour? Any subsequent searches are quick?

The first search being slow and following searches being slow should not be related to this workflow, since this workflow does exactly the same thing on every run. So just the *first* run being slow should either be an issue with Alfred or with your system.
I'd try: 
- updating to the latest version of this workflow
- restarting Alfred
- restarting your system

**Comment by Sinled** [comment]
> Could you elaborate what exactly you mean by "first search"? First search in a given day, in a given hour? Any subsequent searches are quick?

i am not sure of exact time period, but for example when i tried first time in the morning - it took 16 seconds, after that subsequent searches worked fine, then again an hour ago was long search and subsequent searches worked fine.

> updating to the latest version of this workflow

thanks 
i'll try when with new version when it will available in the Gallery.

**Comment by Sinled** [comment]
I've tried 4.1.0 version, and result is the same - long first search. I've tried to add logs to main script and here is result

```
```

**Comment by chrisgrieser** [comment]
Hmm, I found one point which might be responsible for some slowdown. I released version 4.2.0, which might fix the issue:
https://github.com/chrisgrieser/shimmering-obsidian/releases/download/4.2.0/shimmering-obsidian.alfredworkflow

**Comment by Sinled** [comment]
It seems it is working fine now, thanks
